deployer/models/release_safe.go

```diff
@@ -60,122 +60,194 @@ func (release *Release) ValidateSafeRelease(s3c aws.S3API, resources *ReleaseRes
 	}
 
 	if res := safeStr(service.Profile, prevService.Profile); res != nil {
-		return fmt.Errorf("SafeRelease Error(%v): Profile different %v", serviceName, *res)
+		srse.Profile = fmt.Errorf("SafeRelease Error(%v): Profile different %v", serviceName, *res)
 	}
 
 	// 3. ELBs or Target Groups
 	if res := safeUnorderedStrList(service.ELBs, prevService.ELBs); res != nil {
-		return fmt.Errorf("SafeRelease Error(%v): ELBs different %v", serviceName, *res)
+		srse.ELBs = fmt.Errorf("SafeRelease Error(%v): ELBs different %v", serviceName, *res)
 	}
 
 	if res := safeUnorderedStrList(service.TargetGroups, prevService.TargetGroups); res != nil {
-		return fmt.Errorf("SafeRelease Error(%v): TargetGroups different %v", serviceName, *res)
+		srse.TargetGroups = fmt.Errorf("SafeRelease Error(%v): TargetGroups different %v", serviceName, *res)
 	}
 
 	// 5. EBS information
 	if res := safeInt64(service.EBSVolumeSize, prevService.EBSVolumeSize); res != nil {
-		return fmt.Errorf("SafeRelease Error(%v): EBSVolumeSize different %v", serviceName, *res)
+		srse.EBSVolumeSize = fmt.Errorf("SafeRelease Error(%v): EBSVolumeSize different %v", serviceName, *res)
 	}
 
 	if res := safeStr(service.EBSVolumeType, prevService.EBSVolumeType); res != nil {
-		return fmt.Errorf("SafeRelease Error(%v): EBSVolumeType different %v", serviceName, *res)
+		srse.EBSVolumeType = fmt.Errorf("SafeRelease Error(%v): EBSVolumeType different %v", serviceName, *res)
 	}
 
 	if res := safeStr(service.EBSDeviceName, prevService.EBSDeviceName); res != nil {
-		return fmt.Errorf("SafeRelease Error(%v): EBSDeviceName different %v", serviceName, *res)
+		srse.EBSDeviceName = fmt.Errorf("SafeRelease Error(%v): EBSDeviceName different %v", serviceName, *res)
 	}
 
 	// 6. AssociatePublicIpAddress
 	if res := safeBool(service.AssociatePublicIpAddress, prevService.AssociatePublicIpAddress); res != nil {
-		return fmt.Errorf("SafeRelease Error(%v): AssociatePublicIpAddress different %v", serviceName, *res)
+		srse.AssociatePublicIpAddress = fmt.Errorf("SafeRelease Error(%v): AssociatePublicIpAddress different %v", serviceName, *res)
 	}
 
 	// 4. Instance Type
 	if res := safeStr(service.InstanceType, prevService.InstanceType); res != nil {
-		return fmt.Errorf("SafeRelease Error(%v): InstanceType different %v", serviceName, *res)
+		srse.InstanceType = fmt.Errorf("SafeRelease Error(%v): InstanceType different %v", serviceName, *res)
 	}
 
-	if err := validaeSafeAutoscaling(serviceName, service.Autoscaling, prevService.Autoscaling); err != nil {
-		return err
-	}
+	validaeSafeAutoscaling(srse, serviceName, service.Autoscaling, prevService.Autoscaling)
 
-	return nil
+	return srse
 }
 
-func validaeSafeAutoscaling(serviceName string, as *AutoScalingConfig, prevAs *AutoScalingConfig) error {
+func validaeSafeAutoscaling(srse *SafeReleaseServiceError, serviceName string, as *AutoScalingConfig, prevAs *AutoScalingConfig) {
```

Comment:
`validae` again, gotta love the mac butterfly keyboard

Reply:
I too blame all my typos on Apple :)